tensorbay/opendataset/PlantSeedlings/loader.py

```diff
@@ -0,0 +1,202 @@
+
+    """
+    root_path = os.path.abspath(os.path.expanduser(path))
+    dataset = Dataset("PlantSeedlings_Fullimages")
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog_Fullimages.json"))
+    dataset.add_segment(_get_fullimages_segment(root_path))
+    return dataset
+
+
+def _get_fullimages_segment(root_path: str) -> Segment:
+    segment = Segment("Fullimages")
+    files_path = os.path.join(root_path, "Fullimages")
+    category_map = _get_category_map()
+    for file_path in glob(os.path.join(files_path, "[0-9]*")):
+        for data in _generate_fullimages_data(file_path, category_map[os.path.basename(file_path)]):
+            segment.append(data)
+    return segment
+
+
+def _get_category_map() -> Dict[str, str]:
+    category_map = {}
+    for key, value in _FULLIMAGES_MAP.items():
+        for index in range(key[0], key[1] + 1):
+            category_map[str(index)] = value
+    return category_map
+
+
+def _generate_fullimages_data(data_path: str, category: str) -> Iterator[Data]:
+    folder_name = os.path.basename(data_path)
+    for data_sub_path in os.listdir(data_path):
+        data = Data(
+            os.path.join(data_path, data_sub_path),
+            target_remote_path=f"{folder_name.zfill(2)}\\{data_sub_path}",
+        )
+        data.label.classification = Classification(category=category)
+        yield data
+
+
+def _get_segment(root_path: str, subset: str, category_id_map: Dict[str, int]) -> Segment:
+    category_id = category_id_map[subset]
+    segment = Segment(subset.replace(" ", "_"))
+    data_path = os.path.join(root_path, subset)
+    segmented_path = os.path.join(root_path, "Segmented", subset)
+    single_channel_subset_path = os.path.join(root_path, "single_channel_segmented", subset)
+    os.makedirs(single_channel_subset_path, exist_ok=True)
+    for data in _generate_data(data_path, segmented_path, single_channel_subset_path, category_id):
+        segment.append(data)
+    return segment
+
+
+def _generate_data(
+    data_path: str, segmented_path: str, single_channel_subset_path: str, category_id: int
+) -> Iterator[Data]:
+    for image_path in os.listdir(data_path):
+        semantic_mask_path = os.path.join(segmented_path, image_path)
+        if not os.path.exists(semantic_mask_path):
+            # Discard data without mask
+            continue
+        data = Data(
+            os.path.join(data_path, image_path),
+            target_remote_path=image_path.zfill(7),
+        )
+        semantic_mask = _get_semantic_mask(
+            semantic_mask_path, image_path, single_channel_subset_path, category_id
+        )
+        data.label.semantic_mask = SemanticMask(semantic_mask)
+        yield data
+
+
+def _get_semantic_mask(
+    semantic_mask_path: str, image_path: str, single_channel_subset_path: str, category_id: int
+) -> str:
+    mask_path = os.path.join(single_channel_subset_path, image_path)
```

Comment:
will it be better if:
`mask_path` -> `mask_filename`
`image_path` -> `image_filename`?